### PR TITLE
Bug/#7741 - Remove unavailable custom dimension tiles in the view-only dashboard 

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #7741 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/datastore/user/key-metrics.js [hunk 1]
```diff
@@ -351,16 +353,67 @@ const baseSelectors = {
...
-	},
+	getKeyMetricsSettings: createRegistrySelector( ( select ) => ( state ) => {
+		const keyMetricsSettings = state.keyMetricsSettings;
+
+		const isAuthenticated = select( CORE_USER ).isAuthenticated();
+		const canManageOptions = select( CORE_USER ).hasCapability(
+			PERMISSION_MANAGE_OPTIONS
+		);
+
+		const isViewOnly = ! isAuthenticated && ! canManageOptions;
```
Comment (techanvil) [inline assets/js/googlesitekit/datastore/user/key-metrics.js:375]: Hey @hussain-t, are you sure we need to check for the `manage_options` permission here?

Looking at the backend code where we set the `data-view-only` attribute that underpins the `useViewOnly()` hook and so forth, it's only reliant on the authenticated state (well, and the `VIEW_DASHBOARD` permission, but we can assume we're on the Dashboard already here):

https://github.com/google/site-kit-wp/blob/1ee627f779b1ce0ea24580b8bf8f8ecee8670416/includes/Core/Admin/Screens.php#L371-L379

Also, it should be implicit that an authenticated user has the `manage_options` permission as that's what the `AUTHENTICATE` permission maps to:

https://github.com/google/site-kit-wp/blob/1ee627f779b1ce0ea24580b8bf8f8ecee8670416/includes/Core/Permissions/Permissions.php#L150-L152

Maybe I am missing something, but I think we should be able to simplify things a bit by just checking for `isAuthenticated` here...

File: assets/js/googlesitekit/datastore/user/key-metrics.test.js [hunk 2]
```diff
@@ -527,9 +544,68 @@ describe( 'core/user key metrics', () => {
 					coreKeyMetricsEndpointRegExp
 				);
 			} );
+
+			it( 'should return the filtered widget slugs that do not require custom dimensions when the user is in a view-only dashboard', () => {
+				// Set up state to simulate view-only mode
```
Comment (techanvil) [inline assets/js/googlesitekit/datastore/user/key-metrics.test.js:549]: Please finish comments with a full stop :)

```suggestion
				// Set up state to simulate view-only mode.
```

File: assets/js/googlesitekit/datastore/user/key-metrics.test.js [hunk 3]
```diff
@@ -527,9 +544,68 @@ describe( 'core/user key metrics', () => {
...
+					.select( CORE_USER )
+					.getKeyMetricsSettings();
+				expect( keyMetricsSettings.widgetSlugs ).toEqual( [
+					KM_ANALYTICS_NEW_VISITORS,
+					KM_ANALYTICS_PAGES_PER_VISIT,
+				] );
+			} );
+
+			it( 'should return an empty array if only one widget slug is present when the user is in a view-only dashboard', () => {
+				// Set up state to simulate view-only mode
```
Comment (techanvil) [inline assets/js/googlesitekit/datastore/user/key-metrics.test.js:578]: Same here:

```suggestion
				// Set up state to simulate view-only mode.
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Hey @hussain-t, this is looking good, I do have a question about the use of `manage_options`. I've also left a couple of super trivial comments about comments that I would have updated myself if I weren't sending it back with the aforementioned question :)

File: assets/js/googlesitekit/datastore/user/key-metrics.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/googlesitekit/datastore/user/key-metrics.js:375]: That's a good point. Thanks, @techanvil 👍

Comment (hussain-t) [reply assets/js/googlesitekit/datastore/user/key-metrics.js]: Updated ✅

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**